add gradient component (#438)
microsoft/atlas-design: site/src/components/gradients.md

---
title: Gradients
description: Gradient related components in the Atlas Design System
template: standard
---

# Gradients

There are two main types of gradients.

> 1. Gradients for blending colors between elements and contains (`.graident-border-*`).

Reviewer: nit: spelling
```suggestion
1. Gradients for blending colors between elements and contains (`.gradient-border-*`).
```

Author: updated!

2. Gradients displayed with font. Coming soon.

## Gradient borders usage

> The `.gradient-border` component is a moderately pluggable component used to blend the boundaries between colors. Because of the near-infinite number of steps, color variations, and gradient types involved in creating a gradient. We've scope these to be linear gradients that transition between `transparent` and a given color represented by the `--border-gradient-end-color` custom property. Each gradient combination requires a minimum of two class, one specifying the color and one specifying the direcition.

Reviewer: nit: grammar and spelling
```suggestion
The `.gradient-border` component is a moderately pluggable component used to blend the boundaries between colors. Because of the near-infinite number of steps, color variations, and gradient types involved in creating a gradient, we've scoped these to be linear gradients that transition between `transparent` and a given color represented by the `--border-gradient-end-color` custom property. Each gradient combination requires a minimum of two class, one specifying the color and one specifying the direction.
```

Author: updated!

| base class name                  | interpolated value    | screensize |
| -------------------------------- | --------------------- | ---------- | --- |
| `gradient-border-to-<direction>` | `right`, `bottom`     | `tablet`   |

> | `gradient-border-<colorname>`    | `body`, `body-accent` | n/a        |     |

Author: updated!

	>
		A left gradient is applied and a different background color.
	</div>
	<div class="background-color-body-background padding-xl flex-grow-1">
		A background color is applied.
	</div>
</div>
```

## A bottom gradient

The follow shows an example with two elements. The top element contains a bottom gradient that blends toward the bottom element, which has a matching background color class applied. _There is a middle border class applied that is purely used to show the distinction between the two elements._

```html
<div class="gradient-border-bottom gradient-border-body-accent padding-xl">
	A bottom gradient is applied.
</div>
<div class="background-color-body-accent padding-xl">A background color is applied</div>
```

The inverse ...

```html
<div class="gradient-border-bottom background-color-body-accent gradient-border-body padding-xl">
	A bottom gradient is applied.
</div>
<div class="padding-xl ">A background color is applied</div>
```

### Extending gradient directions and colors

By default Atlas has only a few gradient colors and directions to fit a few specific use cases.

> If you need to make a new color available, look to the `$border-gradient-color` Sass map in `src/gradients/components, and add a new color. Alternatively, for downstream projects, specify your own implementation of `$border-gradient-colors` without the default keyword. See [Sass variables documentation](https://sass-lang.com/documentation/variables#advanced-variable-functions) for relevant information.

Reviewer: nit: Markdown fix
```suggestion
If you need to make a new color available, look to the `$border-gradient-color` Sass map in `src/gradients/components`, and add a new color. Alternatively, for downstream projects, specify your own implementation of `$border-gradient-colors` without the default keyword. See [Sass variables documentation](https://sass-lang.com/documentation/variables#advanced-variable-functions) for relevant information.
```

Author: updated!
